packages/astro/src/vite-plugin-astro/hmr.ts

```diff
@@ -91,7 +91,7 @@ export async function handleHotUpdate(
 	// Bugfix: sometimes style URLs get normalized and end with `lang.css=`
 	// These will cause full reloads, so filter them out here
 	const mods = [...filtered].filter((m) => !m.url.endsWith('='));
-	const file = ctx.file.replace(config.root.pathname, '/');
+	const file = ctx.file.replace(fileURLToPath(config.root).replace(/\\/g, '/'), '/');
```

Comment:
Super minor, but we have a few utilities that might be helpful exported from [`@astrojs/internal-helpers/path`](https://github.com/withastro/astro/blob/main/packages/internal-helpers/src/path.ts). Namely [`slash`](https://github.com/withastro/astro/blob/0d4922dffe7d96b907db89ec502f77efd572cc56/packages/internal-helpers/src/path.ts#L92) could replace the manual regex logic here.